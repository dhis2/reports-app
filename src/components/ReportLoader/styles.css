.report-loader {
    margin: 48px 0;
    text-align: center;
}
