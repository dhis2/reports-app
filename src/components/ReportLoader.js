import React from 'react'
import PropTypes from 'prop-types'
import { CircularProgress } from '@dhis2/d2-ui-core'
import isEmpty from 'lodash.isempty'
import './ReportLoader/styles.css'

const ReportLoader = ({ isLoading, content, children }) => {
    if (isLoading) {
        return (
            <div className="report-loader">
                <CircularProgress />
            </div>
        )
    }

    if (isEmpty(content)) {
        return null
    }

    return children
}

ReportLoader.propTypes = {
    isLoading: PropTypes.bool.isRequired,
    content: PropTypes.object.isRequired,
    children: PropTypes.node.isRequired,
}

export default ReportLoader
