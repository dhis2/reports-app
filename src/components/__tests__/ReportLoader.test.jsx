import { shallow } from 'enzyme'
import React from 'react'
import ReportLoader from '../ReportLoader.jsx'

describe('<ReportLoader/>', () => {
    it('renders a CircularProgress when loading', () => {
        const wrapper = shallow(
            <ReportLoader content={{}} isLoading={true}>
                <div>Hello</div>
            </ReportLoader>
        )
        expect(wrapper.find('CircularProgress').exists()).toBe(true)
    })

    it('renders nothing when empty content & not loading', () => {
        const wrapper = shallow(
            <ReportLoader content={{}} isLoading={false}>
                <div>Hello</div>
            </ReportLoader>
        )
        expect(wrapper.type()).toBeNull()
    })

    it('renders children when not loading and content is non-empty', () => {
        const wrapper = shallow(
            <ReportLoader content={{ foo: 'bar' }} isLoading={false}>
                <div>Hello</div>
            </ReportLoader>
        )
        expect(wrapper.contains(<div>Hello</div>)).toBe(true)
    })
})
