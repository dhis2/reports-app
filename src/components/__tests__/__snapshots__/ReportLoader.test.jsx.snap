// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportLoader/> Renders a loader when isLoading is true 1`] = `
<WithStyles(Paper)>
  <div
    className="report-loader"
  >
    <CircularProgress
      large={false}
      small={false}
      style={Object {}}
    />
    <style>
      
                div {
                    margin: 48px 0;
                    text-align: center;
                }
            
    </style>
  </div>
</WithStyles(Paper)>
`;

exports[`<ReportLoader/> Renders its children when isLoading is false and content is non-empty 1`] = `"<div class=\\"MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2\\"><div>Hello world</div></div>"`;
